### PR TITLE
Lot naming fixes

### PR DESCRIPTION
**Pull request by Cosmatevs**
simplified number validation
made "|" a special character
fixed "too many apostrophes/dashes" bug

### CODE REVIEW

File: TSOClient/FSO.Common.Domain/Realestate/RealestateDomain.cs [hunk 1]
```diff
@@ -13,8 +13,11 @@ namespace FSO.Common.Domain.Realestate
 {
     public class RealestateDomain : IRealestateDomain
     {
-        private Regex VALIDATE_NUMERIC = new Regex(".*[0-9]+.*");
-        private Regex VALIDATE_SPECIAL_CHARS = new Regex("[a-z|A-Z|-| |']*");
+        // No need to check redundant regex conditions until you have to throw various errors
+        // (vide tso.client/UI/Panels/UILotPurchaseDialog.cs)
+        // I tried to combine conditions to reduce redundancy
+        private Regex VALIDATE_APOSTROPHES = new Regex("([a-zA-Z -]*)'?(?1)");
```
Comment (riperiperi) [inline TSOClient/FSO.Common.Domain/Realestate/RealestateDomain.cs]: `System.ArgumentException: 'parsing "([a-zA-Z -]*)'?(?1)" - Unrecognized grouping construct.'`

.NET 4.5 does not seem to parse this regex correctly. (.NET regex is somewhat limited) You should verify regex by either testing with C# immediate mode or a simple C# test program.

Comment (Cosmatevs) [reply TSOClient/FSO.Common.Domain/Realestate/RealestateDomain.cs]: I forgot to test those regexes although I was going to do it… I've changed them, they should be compilable now.

File: TSOClient/tso.client/UI/Panels/UILotPurchaseDialog.cs [hunk 2]
```diff
@@ -15,8 +15,11 @@ namespace FSO.Client.UI.Panels
 {
     public class UILotPurchaseDialog : UIDialog
     {
-        private Regex VALIDATE_NUMERIC = new Regex(".*[0-9]+.*");
-        private Regex VALIDATE_SPECIAL_CHARS = new Regex("[a-z|A-Z|-| |']*");
+        private Regex VALIDATE_NUMERIC = new Regex("[0-9]");
+        private Regex VALIDATE_SPECIAL_CHARS = new Regex(@"[^\p{L} '-]");
+        private Regex VALIDATE_APOSTROPHES = new Regex("^[^']*'?[^']*$");
+        private Regex VALIDATE_DASHES = new Regex("^[^-]*-?[^-]*$");
+        private Regex VALIDATE_SPACES = new Regex("^[^ ]+( [^ ]+)?( [^ ]+)?$");
```
Comment (riperiperi) [inline TSOClient/tso.client/UI/Panels/UILotPurchaseDialog.cs]: Same as above.

Comment (Cosmatevs) [reply TSOClient/tso.client/UI/Panels/UILotPurchaseDialog.cs]: I don't know why have I done it that way… But it's fixed now. I've also prevented the regex from creating unnecessary groups.

**Pull request merged**